sandbox-backend/src/main/java/com/exadel/sandbox/team5/dao/DiscountDAO.java

```diff
@@ -2,9 +2,36 @@
 
 import com.exadel.sandbox.team5.entity.Discount;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+import java.util.Set;
+
 @Repository
 public interface DiscountDAO extends JpaRepository<Discount, Long> {
 
+    @Query(value = "SELECT d.*, AVG(r.rate) " +
```

Comment:
Here we should use the following alias `AVG(r.rate)` -> `AVG(r.rate) rate`.

Reply:
done